src/main/java/codechicken/nei/ItemList.java

```diff
@@ -93,15 +93,15 @@ public PatternItemFilter(Pattern pattern) {
         public boolean matches(ItemStack item) {
             String displayName = EnumChatFormatting.getTextWithoutFormattingCodes(item.getDisplayName());
 
-            if (!displayName.isEmpty() && pattern.matcher(displayName).find()) {
+            if (displayName != null && !displayName.isEmpty() && pattern.matcher(displayName).find()) {
```

Comment:
you can also do `"".equals(displayName)` if you want as a shortcut, it will do a null check + check that it is empty

Reply:
hm.... its dont work for me
|||
|-|-|
| ![image](https://github.com/user-attachments/assets/68dec933-fea0-4e20-9e0e-7b66ef47272a) | ![image](https://github.com/user-attachments/assets/1842964b-2ce5-4052-bc51-31f4dd3a45fc) |

Reply:
nvm I'm dumb, it's `text != null && text.isEmpty()` that's equivalent to doing `"".equals(text)` but here that's not what we want to do